### ISSUE TITLE
cleanbots cause gamebreaking amounts of lag

### ISSUE THREAD

**Issue by OneOneThreeEight** [Aurorastation/Aurora.3#225]

**Comment by NanakoAC** [comment]
As far as i can tell, they get stuck in some sort of loop with pathfinding, and will bring the server to a crawl until they're either turned off, or destroyed

**Comment by skull132** [comment]
```
Infinite loop suspected--switching proc to background.
If it is not an infinite loop, either do 'set background=1' or set world.loop_checks=0.
proc name: Life (/mob/living/bot/cleanbot/Life)
  source file: cleanbot.dm,128
  usr: null
  src: Cleanbot (/mob/living/bot/cleanbot)
  src.loc: the floor (173,121,1) (/turf/simulated/floor)
  call stack:
Cleanbot (/mob/living/bot/cleanbot): Life()
/datum/updateQueueWorker (/datum/updateQueueWorker): doWork()
/datum/updateQueueWorker (/datum/updateQueueWorker): doWork()
```

There we gooo.

**Comment by NanakoAC** [comment]
Progress report?

**Comment by skull132** [comment]
Cleanbots now auto-destroy upon spawn. *nodnod

**Comment by OneOneThreeEight** [comment]
There is a greater necessity for janitors anyway.

**Comment by skull132** [comment]
Fixed in #891
